app/controllers/posts_controller.rb

```diff
@@ -2,24 +2,26 @@ class PostsController < ApplicationController
   include ActionView::RecordIdentifier
 
   def index
-    @posts = Post.all
     @users = User.find_by(id: params[:user_id])
-    puts @posts
-  end
+    @users = User.new if @users.nil?
+    @posts = @users.posts
```

Comment:
- [ ] Please address the N+1 problem when requesting all the posts for a user. ensure that you eagerly load the comments of the post when querying for the posts.